app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -14,32 +14,43 @@ def google_oauth2
     end
   end
 
+  def facebook
+    if user.present?
+      user.accept_invitation! if invitation_token
```

Comment:
The `context 'when third party avatar is not nil'` is failing because we are not uploading the `third_party_avata` in the `facebook` OAuth action.

Right below this line, we shall have to add `upload_avatar(omniauth_avatar)`. Reference the `google_oauth2` action at [line 7](https://github.com/ifmeorg/ifme/pull/1705/files#diff-aed97266b0fe72994225a9cac3e4230bR7)